dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04315UpdateMultiTreePK.java

```diff
@@ -53,8 +54,8 @@ public boolean forceRun() {
     public void executeUpgrade() throws DotDataException {
         
         try {
-            Connection conn =DbConnectionFactory.getConnection();
-            
+            Connection conn = DbConnectionFactory.getDataSource().getConnection();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "Connection" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)